apps/condo/cypress/objects/helpers.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Similar to cy.visit(url), but will track page loading time.
+ * @param {string} url
+ * @example trackedVisit('/auth/signin/')
+ */
+const trackedVisit: (url) => Cypress.Chainable<Performance> = (url) => {
+    // transforms /auth/signin/ to auth.signin
+    // if empty: then 'index' is chosen as metric name
+    const normalizedUrl = url.replace(/^\/?|\/?$/g, '').replaceAll('/', '.') || 'index'
+
+    return cy.visit(url, {
+        onBeforeLoad: (win) => {
+            win.performance.mark(`${normalizedUrl}.startLoading`)
+        },
+        onLoad: (win) => {
+            win.performance.mark(`${normalizedUrl}.stopLoading`)
+        },
+    }).its('performance').then((p) => {
+        p.measure(`${normalizedUrl}.pageLoad`, `${normalizedUrl}.startLoading`, `${normalizedUrl}.stopLoading`)
+        const measure = p.getEntriesByName(`${normalizedUrl}.pageLoad`)[0]
```

Comment:
why 0 ?

Reply:
https://udn.realityripple.com/docs/Web/API/Performance/getEntriesByName

getEntriesByName returns an array of performanceEntries, that are created by `mark()` function. These objects are not unique by design, so they return array here

Since we have unique PerformanceEntry objects here, we use [0]

Reply:
Add some comment